client/client_test.go

```diff
@@ -0,0 +1,256 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// The IP adress of the test server
+const TestServerIP = "127.0.0.1"
+
+// The port that the server is listening on
+const TestServerPort = 8022
+
+// startTestServer starts a server listening at `TestServerIP` on
+// `TestServerPort`.  Need to later manually tear down the server.
+func startTestServer() (int, string, error) {
+	dir, err := ioutil.TempDir("", "TestServer")
+	if err != nil {
+		return 0, "", err
+	}
+
+	cmd := exec.Command("go",
```

Comment:
hmm to set up the fake server, I would need to implement the entire interface, which includes SearchWord. I am planning to use a local map to store info for the fake server, but SearchWord would still be basically the same. Should I just copy the code for SearchWord over as the "search-server" repo is private?